packages/flutter/test/widgets/slivers_padding_test.dart

```diff
@@ -618,7 +618,16 @@ void main() {
   });
 }
 
-void expectIsTextSpan(Object target, String text) {
-  expect(target, isA<TextSpan>());
-  expect((target as TextSpan).text, text);
+void hitsText(HitTestResult hitTestResult, String text) {
+  switch (hitTestResult.path.first.target) {
+    case final TextSpan span:
+      expect(span.text, text);
+    case final RenderParagraph paragraph:
+      final InlineSpan span = paragraph.text;
+      expect(span, isA<TextSpan>());
+      expect((span as TextSpan).text, text);
+    case final HitTestTarget target:
+      // This expect always fails.
+      expect(target, anyOf(isA<TextSpan>(), isA<RenderParagraph>()));
```

Comment:
Why not use `fail` then to make it explicit that this is a failure case and provide a failure message?

https://main-api.flutter.dev/flutter/package-matcher_expect/fail.html